init.lua

```diff
@@ -167,10 +167,13 @@ end
 
 -- mapping function between vis lexer names and LSP languageIds
 local function syntax_to_languageId(syntax)
-  if syntax == 'ansi_c' then
-    return 'c'
-  end
-  return syntax
+  local map = {
```

Comment:
This does not seem correct to me. What is the reason for this?

[List of language IDs](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentItem)

* JavaScript uses `javascript` not `jsx`
* TypeScript uses `typescript` not `tsx`

Reply:
Yeah it's a workaround for `jsx/tsx` files which are a superset of javascript/typescript allowing for xml-like syntax for creating elements (used in react mostly). So rather than launching separate language servers for a ts and tsx file we just use one.

See also https://github.com/kakoune-lsp/kakoune-lsp/issues/211

Reply:
My problem with this change is that. Someone configuring their setup may not know about the enforced mapping and tries to configure a language server for the correct syntax/languageID `javascript`, which will not be picked up due to our mapping.

Maybe we can come up with a more generic solution for this.

Our solution should support setups that want to map all JS/TS files to JSX/TSX, ones that allow to separate both and ones that only want to configure servers for JS/TSX.

Reply:
Hm, not sure of a clean solution as vis's filetype plugin only exposes `*.tsx` files as typescript files

And as I said it's a superset and I don't think there's any other js/ts language server in active development that doesn't support jsx/tsx

Reply:
Applied as a2dc6b94c97dcb55d5451fcf7b89498dbc0b9b9d.
 
 My initial concern is invalid since we use the mapping only to determine the `languageId` passed to the server not to select a language server configuration.
 
 We can still think about a more generic/complex solution if someone encounters an issue with this mapping.